internal/manifests/alerts/prometheus-rules.yaml

```diff
@@ -2,15 +2,15 @@
 groups:
 - name: "tempo_rules"
   rules:
-  - expr: "histogram_quantile(0.99, sum(rate(tempo_request_duration_seconds_bucket[1m])) by (le, cluster, namespace, job, route))"
-    record: "cluster_namespace_job_route:tempo_request_duration_seconds:99quantile"
-  - expr: "histogram_quantile(0.50, sum(rate(tempo_request_duration_seconds_bucket[1m])) by (le, cluster, namespace, job, route))"
-    record: "cluster_namespace_job_route:tempo_request_duration_seconds:50quantile"
-  - expr: "sum(rate(tempo_request_duration_seconds_sum[1m])) by (cluster, namespace, job, route) / sum(rate(tempo_request_duration_seconds_count[1m])) by (cluster, namespace, job, route)"
-    record: "cluster_namespace_job_route:tempo_request_duration_seconds:avg"
-  - expr: "sum(rate(tempo_request_duration_seconds_bucket[1m])) by (le, cluster, namespace, job, route)"
-    record: "cluster_namespace_job_route:tempo_request_duration_seconds_bucket:sum_rate"
-  - expr: "sum(rate(tempo_request_duration_seconds_sum[1m])) by (cluster, namespace, job, route)"
-    record: "cluster_namespace_job_route:tempo_request_duration_seconds_sum:sum_rate"
-  - expr: "sum(rate(tempo_request_duration_seconds_count[1m])) by (cluster, namespace, job, route)"
-    record: "cluster_namespace_job_route:tempo_request_duration_seconds_count:sum_rate"
+  - expr: "histogram_quantile(0.99, sum(rate(tempo_request_duration_seconds_bucket{cluster=\"[[ .Cluster ]]\", namespace=\"[[ .Namespace ]]\"}[1m])) by (le, cluster, namespace, job, route))"
+    record: "[[ .Cluster ]]_[[ .Namespace ]]:cluster_namespace_job_route:tempo_request_duration_seconds:99quantile"
```

Comment:
Is it required to update the name (`record`) of the new metrics?

In my cluster I have the old rules (before this PR) and I see the following metrics:
```
cluster_namespace_job_route:tempo_request_duration_seconds:99quantile{cluster="simplest", job="default/compactor", namespace="default", route="metrics"}
```
So the recorded metrics already contain the `namespace` and `cluster` labels, so I *think* (I didn't test it) it'd be fine to only change the `expr` (as you did) but leave the `record` the same, wdyt?

Reply:
If I don't scope the `records`, those are overridden

If I create a "simplest" `Tempostack`, this is how the record looks like: 
![Screenshot from 2023-07-17 22-22-41](https://github.com/grafana/tempo-operator/assets/3799716/f2f0d468-c472-4ead-a15b-70ba5da119f1)

Then after create simplest2, this is the only records appears on Prometheus.

![Screenshot from 2023-07-17 22-20-42](https://github.com/grafana/tempo-operator/assets/3799716/9dcee91b-3c7a-4652-abc3-5efd66181eee)

Reply:
I cannot reproduce this behavior - I just tried it on my cluster (using Prometheus 2.44.0), when I open the `/rules` page I see two sections `tempo_alerts` and two sections for `tempo_rules` (one with `simplest`, one with `simplest2`).

When I query for a recorded metric, for example `cluster_namespace_job_route:tempo_request_duration_seconds:99quantile`, I see values for both TempoStack instances: ![Bildschirmfoto vom 2023-07-18 13-00-49](https://github.com/grafana/tempo-operator/assets/538011/bb4e75ab-7b43-4712-83f7-4a9612865516)

Can you check this page again to be sure you don't have two `tempo_rules` sections? (fwiw, maybe we could also rename `tempo_rules` and `tempo_alerts` to `tempo_rules_$cluster_$namespace` and `tempo_alerts_$cluster_$namespace`?)

Reply:
I was able to reproduce what you have, I was confused by the fact the groups are the same, but I have a question about what happens when you refer a record in the alerts section and two records has the same name?